pkg/kubelet/images/image_manager.go

```diff
@@ -73,19 +73,40 @@ func NewImageManager(recorder record.EventRecorder, imageService kubecontainer.I
 	}
 }
 
-// shouldPullImage returns whether we should pull an image according to
-// the presence and pull policy of the image.
-func shouldPullImage(pullPolicy v1.PullPolicy, imagePresent bool) bool {
-	if pullPolicy == v1.PullNever {
-		return false
-	}
-
-	if pullPolicy == v1.PullAlways ||
-		(pullPolicy == v1.PullIfNotPresent && (!imagePresent)) {
-		return true
-	}
-
-	return false
+// imagePullPrecheck inspects the pull policy and checks for image presence accordingly,
+// returning (imageRef, error msg, err) and logging any errors.
+func (m *imageManager) imagePullPrecheck(ctx context.Context, objRef *v1.ObjectReference, logPrefix string, pullPolicy v1.PullPolicy, spec *kubecontainer.ImageSpec, imgRef string) (imageRef string, msg string, err error) {
+	switch pullPolicy {
+	case v1.PullAlways:
+		return "", msg, nil
+	case v1.PullIfNotPresent:
+		imageRef, err = m.imageService.GetImageRef(ctx, *spec)
+		if err != nil {
+			msg = fmt.Sprintf("Failed to inspect image %q: %v", imageRef, err)
+			m.logIt(objRef, v1.EventTypeWarning, events.FailedToInspectImage, logPrefix, msg, klog.Warning)
+			return "", msg, ErrImageInspect
+		}
+		return imageRef, msg, nil
+	case v1.PullNever:
+		imageRef, err = m.imageService.GetImageRef(ctx, *spec)
+		if err != nil {
+			msg = fmt.Sprintf("Failed to inspect image %q: %v", imageRef, err)
+			m.logIt(objRef, v1.EventTypeWarning, events.FailedToInspectImage, logPrefix, msg, klog.Warning)
+			return "", msg, ErrImageInspect
+		}
+		if imageRef == "" {
+			msg = fmt.Sprintf("Container image %q is not present with pull policy of Never", imgRef)
+			m.logIt(objRef, v1.EventTypeWarning, events.ErrImageNeverPullPolicy, logPrefix, msg, klog.Warning)
+			return "", msg, ErrImageNeverPull
+		}
+		return imageRef, msg, nil
+	default:
```

Comment:
If you add this `default`, the `return` at the end of the function will never be executed. 
Additionally, I believe this `default` is unnecessary, because the `Apiserver` will not accept any string as `imagePullPolicy` that is not one of the enumerated values.

Reply:
Removed it!